LoRaEngine/LoraKeysManagerFacade/LoRaDevAddrCache.cs

```diff
@@ -277,7 +279,7 @@ private void BulkSaveDevAddrCache(List<DevAddrCacheInfo> devAddrCacheInfos, bool
         private static IDictionary<string, DevAddrCacheInfo> KeepExistingCacheInformation(HashEntry[] cacheDevEUIEntry, IGrouping<DevAddr, DevAddrCacheInfo> newDevEUIList, bool canDeleteExistingDevice)
         {
             // if the new value are not different we want to ensure we don't save, to not update the TTL of the item.
-            var toSyncValues = newDevEUIList.ToDictionary(x => x.DevEUI);
+            var toSyncValues = newDevEUIList.ToDictionary(x => GenerateKey(x.DevEUI.Value));
```

Comment:
why is this additional GenerateKey necessarey?

Reply:
Currently, the underlying `ILoRaDeviceCacheStore` only supports keys of type `string`. We could make the underlying support generic keys, but I would propose to do maybe do that in a separate PR to not increase the amount of diffs further.

Reply:
> but I would propose to do maybe do that in a separate PR to not increase the amount of diffs further.

I think it's a little too late for that now. 😉.

Reply:
Ok. Could we simply use ToString() as the default then?

Reply:
nit comment - we can also keep that. Up to you.

Reply:
As discussed, removed with 62362a4ab925d921476280de8601414a3c6ee5f5